packages/component-library/stories/ScreenGridMap.story.js

```diff
@@ -1,83 +1,99 @@
 import React from 'react';
 /* eslint-disable import/no-extraneous-dependencies */
 import { storiesOf } from '@storybook/react';
-import { withKnobs, number, selectV2, boolean } from '@storybook/addon-knobs';
-import { action } from '@storybook/addon-actions';
+import { withKnobs, number, selectV2 } from '@storybook/addon-knobs';
 import { checkA11y } from '@storybook/addon-a11y';
 import { BaseMap } from '../src';
 import { ScreenGridMap } from '../src';
-
-import fireHydrants from '../src/ScreenGridMap/data';
+import * as d3 from 'd3';
 
 const displayName = ScreenGridMap.displayName || 'ScreenGridMap';
 
-const mapboxToken = 'pk.eyJ1IjoidGhlbWVuZG96YWxpbmUiLCJhIjoiY2o1aXdoem1vMWtpNDJ3bnpqaGF1bnlhNSJ9.sjTrNKLW9daDBIGvP3_W0w';
+class LoadData extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    let cmp = this;
+    d3.json(this.props.url, (error, data) => {
+      if (error) { return this.setState({error: error})}
+      cmp.setState({data: data});
```

Comment:
There's no need to store the component in a variable outside this callback since you're using an arrow function. `this.setState({ data });` works.

Reply:
That's what I thought too, but when I didn't include that I kept getting an error that `this` was undefined.